app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb

```diff
@@ -76,17 +76,28 @@ def parse_physical_server(node)
         :model                  => node.model,
         :serial_number          => node.serialNumber,
         :field_replaceable_unit => node.FRU,
-        :computer_system        => {
-          :hardware             => {
-            :networks  => [],
-            :firmwares => [] # Filled in later conditionally on what's available
-          }
-        }
+        # Filled in later conditionally on what's available
+        :computer_system        => {:hardware => {:networks => [], :firmwares => []}}
       }
       new_result[:computer_system][:hardware] = get_hardwares(node)
+      save_host_relationship(node)
       return node.uuid, new_result
     end
 
+    def save_host_relationship(node)
```

Comment:
I think you do the look up and assignments in the refresher as you have already implemented, but you remove the line that saves the server instance.     I think that the existing code in save_ems_physical_infra_inventory (https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh/save_inventory_physical_infra.rb#L8 )  should save the details without any additional changes.  This needs to be verified.